modules/social_features/social_user/src/Form/SocialUserLoginForm.php

```diff
@@ -117,13 +117,12 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $account = $this->userStorage->load($form_state->get('uid'));
     // A destination was set, probably on an exception controller,.
     // @TODO: Add validation if route exists.
-    if (!$this->getRequest()->request->has('destination')) {
+    if (!$this->getRequest()->server->has('REDIRECT_QUERY_STRING')) {
       $form_state->setRedirect('<front>');
     }
     else {
-      $this->getRequest()->query->set('destination', $this->getRequest()->request->get('destination'));
+      $this->getRequest()->query->set('destination', parse_url(urldecode(str_replace('destination=', '', $this->getRequest()->server->get('REDIRECT_QUERY_STRING'))), PHP_URL_PATH));
```

Comment:
I'm wondering: are there better ways to do this?

The way we retrieve the destination right now doesn't seem to be the way to go as it's always better to use Drupal's functions for this. Why doesn't `$this->getRequest()->request->get('destination')` work? Perhaps when we realise this we can find a different solution and still use that method to retrieve the current destination.

Reply:
When debugging the stuff I had no values in:
$this->getRequest()->**request**->get('destination') 
that's why I switched over from the request-scope to the server-scope
$this->getRequest()->**server**->get('REDIRECT_QUERY_STRING')

Reply:
What do you think about the latest changes? Transport destination information in a hidden form field. This keeps other stuff simple and easy to understand.